packages/rln/src/contract/rln_contract.ts

```diff
@@ -275,15 +305,21 @@ export class RLNContract {
         if (!evt.args) return;
 
         const _idCommitment = evt.args.idCommitment as string;
-        const index = evt.args.index as ethers.BigNumber;
+        let index = evt.args.index;
 
         if (!_idCommitment || !index) {
           return;
         }
 
+        if (typeof index === "number" || typeof index === "string") {
```

Comment:
what if it is not any of those?

Reply:
shouldn't be the case as contract events will emit these types; still added an `else` block to invalidate wrong types